h/views/activity.py

```diff
@@ -349,6 +349,19 @@ def domain(user):  # pragma: no cover
             "orcid": self.user.orcid,
         }
 
+        if not self.request.user and (self.user.nipsa or annotation_count == 0):
+            # To avoid linking to potentially spammy or NIPSA'd user profiles
+            # we don't link to the user profile if the user is not logged in
+            # We redirect to the login page instead
+            return httpexceptions.HTTPSeeOther(
+                location=self.request.route_url(
+                    "login",
+                    next=self.request.route_url(
+                        "activity.user_search", username=self.user.username
+                    ),
```

Comment:
I'm not seeing the `next` parameter set in the URL when I test this locally.

Reply:
Oh my bad, I was testing this and if was taking to **a** profile page next, of course I was taken to the user's **own** profile instead of the one I was trying before. I didn't notice the devdata_user vs devdata_admin difference...

Fixed.